src/components/icons/LanguageIcon.tsx

```diff
@@ -0,0 +1,33 @@
+import React from 'react'
+import JavaScriptLogoColour from './JavaScriptLogoColour'
+import TypeScriptLogoColour from './TypeScriptLogoColour'
+import PythonColorLogo from './PythonLogoColour'
+import GoColorLogo from './GoLogoColour'
+import DartLogoNoTextColour from './DartLogoNoTextColour'
+import { cn } from '@/lib/utils'
+
+interface LanguageIconProps {
+  name: string
+  className?: string
+}
+
+const icons: Record<string, React.FC<{ className: string }>> = {
+  javascript: JavaScriptLogoColour,
+  typescript: TypeScriptLogoColour,
+  python: PythonColorLogo,
+  go: GoColorLogo,
+  dart: DartLogoNoTextColour,
+}
+
+export const LanguageIcon: React.FC<LanguageIconProps> = ({
+  name,
+  className,
+}) => {
+  const IconComponent = icons[name]
+
+  if (!IconComponent) {
+    return <span>Unknown language</span>
+  }
```

Comment:
Can you find a way to make this a type error or build error, instead of a silent runtime error?

Reply:
e.g. make the name prop a value from the languages array and add a loop in here that tries to create each one and fails if an icon is missing from the map